### PR TITLE
Completed ds-prep-python-loops.| Rakesh

### PR DESCRIPTION
**Pull request by bhaisingh**
Screenshot from 2022-01-12 03-46-28

### CODE REVIEW

File: ds-prep-python-loops/main.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+record = (1, 'Grimdiana', 'Bones', 'boulders')
+row = ''
+for field in record:
+    row += str(field) + ', '
+print(row)
+
+value_list = [1, 1, 2, 3, 5, 8, 13, 21, 34]
+for value in value_list:
+    print(str(value) + '\n')
```
Comment (TimelikeClosure) [inline ds-prep-python-loops/main.py]: You don't need to add your own newline character here, it results in a bunch of blank lines:

---

image

File: ds-prep-python-loops/main.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+index_list = ''
+for index in range(len(value_list)):
+    print(index, value_list[index])
```
Comment (TimelikeClosure) [inline ds-prep-python-loops/main.py]: `index_list` should be a `list` of the indexes in `value_list`:

---

image

File: ds-prep-python-loops/main.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+for value in value_list:
+    if value % 2 != 0:
+        value_list.remove(value)
```
Comment (TimelikeClosure) [inline ds-prep-python-loops/main.py]: This should loop over the values in `index_list`:

---

image

File: ds-prep-python-loops/main.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "What": "2B",
+    "I Don't Know": "3B",
+    "Why": "LF",
+    "Because": "CF",
+    "Tomorrow": "P",
+    "Today": "C",
+    "I Don't Care": "SS"
+}
+
+players = ''
```
Comment (TimelikeClosure) [inline ds-prep-python-loops/main.py]: `players` should be a `list`, not a `str`:

---

image

File: ds-prep-python-loops/main.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+for keys in player_positions:
+    players += keys + ': ' + player_positions[keys] + '\n'
+print('players:', players)
+
+positions = ''
```
Comment (TimelikeClosure) [inline ds-prep-python-loops/main.py]: `positions` should be a `set`, not a `str`:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by bhaisingh** [comment]
Screenshot from 2022-01-12 19-58-39

File: ds-prep-python-loops/main.py (hunk 1 above)
Comment (bhaisingh) [reply ds-prep-python-loops/main.py]: Removed the new line character

File: ds-prep-python-loops/main.py (hunk 2 above)
Comment (bhaisingh) [reply ds-prep-python-loops/main.py]: Fixed Empty list and loop on Value_list

File: ds-prep-python-loops/main.py (hunk 3 above)
Comment (bhaisingh) [reply ds-prep-python-loops/main.py]: corrected the loop from value_list to index_list

File: ds-prep-python-loops/main.py (hunk 4 above)
Comment (bhaisingh) [reply ds-prep-python-loops/main.py]: corrected assignment of empty list

File: ds-prep-python-loops/main.py (hunk 5 above)
Comment (bhaisingh) [reply ds-prep-python-loops/main.py]: Corrected creation of Empty set and assigning to positions

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**